qlib/kernel/Kernel_cc.rs

```diff
@@ -0,0 +1,2083 @@
+// Copyright (c) 2021 Quark Container Authors / 2018 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::mem::size_of;
+
+use self::range::Range;
+use alloc::boxed::Box;
+
+use super::super::common::*;
+use super::super::config::*;
+use super::super::linux_def::*;
+use super::super::loader::*;
+use super::super::qmsg;
+use super::super::qmsg::*;
+use super::super::socket_buf::*;
+use super::super::tsot_msg::*;
+use super::super::*;
+use super::Kernel::*;
+use crate::kernel_def::HyperCall64;
+use crate::qlib::control_msg::ControlMsg;
+use crate::qlib::nvproxy::frontend_type::RMAPIVersion;
+use crate::qlib::proxy::*;
+use crate::GLOBAL_ALLOCATOR;
+use crate::GUEST_HOST_SHARED_ALLOCATOR;
+
+extern "C" {
+    pub fn rdtsc() -> i64;
+}
+
+impl HostSpace {
+    pub fn LoadProcessKernel_cc(processAddr: u64) -> i64 {
+        let mut new_process = Box::new_in(Process::default(), GUEST_HOST_SHARED_ALLOCATOR);
```

Comment:
@yaoxin1995 A new vec will not apply for memory after data is pushed in, and the Process::default() is a struct with no data inside derived by the default trait. When returning to qvisor, it will resize the vecs inside when pushing data, using its global allocator, which is the shared one.